doc-deploy-stable/action.yml

```diff
@@ -1,164 +1,286 @@
-  checkout:
-      description: 'Whether to checkout the repository or not.'
-      default: true
-      required: false
-      type: boolean
 
 runs:
   using: "composite"
   steps:
 
-    - name: "Checkout project in the GitHub Pages branch"
-      uses: actions/checkout@v3
-      if: ${{ inputs.checkout == 'true' }}
-      with:
-        ref: 'gh-pages'
+    # Collect the version number from the tag or the branch name according to
+    # the reference type. Tags names are expected to follow 'v<MAJOR>.<MINOR>'
+    # or 'v<MAJOR>.<MINOR>.<PATCH>'. Braches must follow
+    # 'release/<MAJOR>.<MINOR>'.
 
-    - name: "List gh-pages content"
+    - name: "Collect version number from the tag"
       shell: bash
+      if: github.ref_type == 'tag'
       run: |
-        ls -R
+        version=$( echo $tag | cut -d 'v' -f 2)
 
-    - name: "Check adequate version tagging"
+        if [[ $version =~ ^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)$ ]]; then
+          echo "VERSION_RAW=$(echo $version)" >> $GITHUB_ENV
+        else
+          echo "[PYANSYS ACTIONS ERROR]"
+          echo "Tag names must follow 'v<MAJOR>.<MINOR>.<PATCH> name convention.'"
+          exit 1
+        fi
+
+    - name: "Collect version number from the release branch"
       shell: bash
+      if: github.ref_type == 'branch'
       run: |
-        # Remove the 'v' or the 'release/' at the beginning of the tag
-        ref_name="${{ github.ref_name }}"
-        version=${ref_name##*v}
-        version=${version##*release/}
-        # HACK: pydfp uses release-X.Y.Z
-        version=${version##*release-}
-        # Check that the version is of the accepted format
-        echo "ACCEPTED_FORMAT=$([[ "$version" =~ ^[0-9]+\.[0-9]+$|^[0-9]+\.[0-9]+\.[0-9]+$ ]] && echo "true" || echo "false")" >> $GITHUB_ENV
-        echo "PRELIMINARY_VERSION=$version" >> $GITHUB_ENV
+        version=$( echo $tag | cut -d 'release/' -f 2)
+
+        if [[ $version =~ ^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)$ ]]; then
+          echo "VERSION_RAW=$(echo $version)" >> $GITHUB_ENV
+        else
+          echo "[PYANSYS ACTIONS ERROR]"
+          echo "Branch names must follow 'release/<MAJOR>.<MINOR>' name convention."
+          exit 1
+        fi
 
-    - name: "Ignoring deployment... invalid version format!"
-      if: env.ACCEPTED_FORMAT == 'false'
+    - name: "Assemble the version number using the collected major and minor numbers"
       shell: bash
       run: |
-        echo "INVALID VERSION FORMAT - $PRELIMINARY_VERSION"
-        echo "Should be of type X.Y or X.Y.Z... Ignoring documentation deployment."
+          major=$(echo $version | cut -d . -f 1)
+          minor=$(echo $version | cut -d . -f 2)
+          echo "VERSION=$major.$minor" >> $GITHUB_ENV
 
-    - name: "Prepare destination folder"
-      if: env.ACCEPTED_FORMAT == 'true'
+    # Checkout the repository branch for deploying the documentation. If this
+    # step fails, then it means that the provided token is not valid.
```

Comment:
It would be great to have this information printed in the log when an error occur.
I know we talked about improving those error messages.
I don't think we have handle it yet.
So just to keep in mind.

Reply:
Related to https://github.com/pyansys/actions/issues/142. Also, I opened https://github.com/pyansys/actions/issues/171.

If possible, let us address those in a different issue now that the whole logic has been verified.

Reply:
Yeah please... This PR starts getting big already